plugins/needs_restarting.py

```diff
@@ -277,15 +279,24 @@ def run(self):
         NEED_REBOOT.extend(opt)
         if self.opts.reboothint:
             need_reboot = set()
+            need_reboot_depends_on_dbus = set()
             installed = self.base.sack.query().installed()
             for pkg in installed.filter(name=NEED_REBOOT):
                 if pkg.installtime > process_start.boot_time:
                     need_reboot.add(pkg.name)
-            if need_reboot:
+
+            dbus_installed = installed.filter(name=['dbus', 'dbus-daemon', 'dbus-broker'])
+            if len(dbus_installed) != 0:
+                for pkg in installed.filter(name=NEED_REBOOT_DEPENDS_ON_DBUS):
+                    if pkg.installtime > process_start.boot_time:
+                        need_reboot_depends_on_dbus.add(pkg.name)
+            if need_reboot or need_reboot_depends_on_dbus:
                 print(_('Core libraries or services have been updated '
                         'since boot-up:'))
                 for name in sorted(need_reboot):
                     print('  * %s' % name)
+                for name in sorted(need_reboot_depends_on_dbus):
+                    print('  * %s (depends on dbus that needs reboot)' % name)
```

Comment:
zlib is a dependency of dbus, therefore the message is somehow cryptic.

Reply:
What about:
`'  * %s (dependency of dbus that needs reboot)'`

Reply:
`'  * %s (dependency of dbus. Recommending to reboot dbus)'`

Reply:
@jrohel had the same perplexity.
would you rather add zlib to the list of printed packages?

Reply:
> '  * %s (dependency of dbus. Recommending to reboot dbus)'

This is more clear

Reply:
`'  * %s (dependency of dbus. Recommending reboot of dbus)'`

Reply:
From my point of view. Computer can be **rebooted**. But dbus is a service that can be **restarted**.

If I understand correctly what is happening,  upgrading zlib will cause the dbus service to restart. And it is recomended to reboot computer. Am I right?

Reply:
correct